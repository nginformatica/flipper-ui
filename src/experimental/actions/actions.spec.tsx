import * as React from 'react'
import { render, screen } from '@testing-library/react'
import { Actions } from '.'

describe('Actions', () => {
    it('should render', () => {
        render(<Actions onConfirm={jest.fn()} />)

        const actionCancel = screen.getByText('Cancelar')
        const actionConfirm = screen.getByText('Confirmar')

        expect(actionCancel).toBeDefined()
        expect(actionConfirm).toBeDefined()
    })

    it('should render with custom labels', () => {
        render(
            <Actions
                labels={{ cancel: 'Voltar', confirm: 'Excluir' }}
                onConfirm={jest.fn()}
            />
        )

        const actionCancel = screen.getByText('Voltar')
        const actionConfirm = screen.getByText('Excluir')

        expect(actionCancel).toBeDefined()
        expect(actionConfirm).toBeDefined()
    })

    it('should render only confirm', () => {
        render(<Actions buttons={['confirm']} onConfirm={jest.fn()} />)

        const actionConfirm = screen.getByText('Confirmar')
        const actionCancel = screen.queryByText('Cancelar')

        expect(actionConfirm).toBeDefined()
        expect(actionCancel).toBeNull()
    })

    it('should render with custom names', () => {
        render(
            <Actions
                names={{ cancel: 'cancelName', confirm: 'confirmName' }}
                onConfirm={jest.fn()}
            />
        )

        const actionCancel = screen.getByTestId('cancel-action')
        const actionConfirm = screen.getByTestId('confirm-action')

        expect(actionCancel).toHaveProperty('name', 'cancelName')
        expect(actionConfirm).toHaveProperty('name', 'confirmName')
    })

    it('should call onCancel', () => {
        const onCancel = jest.fn()

        render(<Actions onCancel={onCancel} onConfirm={jest.fn()} />)

        const actionCancel = screen.getByText('Cancelar')

        actionCancel.click()

        expect(onCancel).toHaveBeenCalled()
    })

    it('should call onConfirm', () => {
        const onConfirm = jest.fn()

        render(<Actions onConfirm={onConfirm} />)

        const actionConfirm = screen.getByText('Confirmar')

        actionConfirm.click()

        expect(onConfirm).toHaveBeenCalled()
    })
})
