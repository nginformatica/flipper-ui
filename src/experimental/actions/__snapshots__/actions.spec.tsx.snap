// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions should match snapshot 1`] = `
<div>
  <div
    align="flex-end"
    class="sc-dZodDb uKEcO"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      data-testid="cancel-action"
      name="cancel-action"
      style="margin: 0px 16px; opacity: 1;"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Voltar
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      data-testid="confirm-action"
      name="confirm-action"
      style="opacity: 1;"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Excluir
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
